classification/classification_network.py

```diff
@@ -241,108 +230,132 @@ def createClassifierRegion(network, classifierType, classifierParams, prevRegion
     # Add new region class to the network
     network.registerRegion(SequenceClassifierRegion)
     PY_REGIONS.append(classifierType.split(".")[1])
-  
+
   # Create the classifier region.
   classifierRegion = network.addRegion(
-      "classifier", classifierType, json.dumps(classifierParams))
+    "classifier", classifierType, json.dumps(classifierParams))
 
   # Disable learning for now (will be enabled in a later training phase)... why???
   classifierRegion.setParameter("learningMode", False)
-  
+
   # Okay to always leave inference mode on; only there for some corner cases.
   classifierRegion.setParameter("inferenceMode", True)
 
+  if classifyPredictedActiveCells:
+    classifierRegion.setParameter("classifyPredictedActiveCells", True)
+  else:
+    classifierRegion.setParameter("classifyPredictedActiveCells", False)
+
   return classifierRegion
 
 
-def createRegions(network, args):
-  """
-  Create the regions. @param args is to hold network params.
-  Note the regions still need to be linked appropriately in linkRegions().
-
-   @param network (Network)    The network instance
-   @param args                 (dataSource, sensorType, encoders, numCategories, 
-                                classifierType, classifierParams) , more info:
-    dataSource   (RecordStream) Sensor region reads data from here.
-    sensorType   (str)          Specific type of region, e.g. "py.RecordSensor";
-                                possible options can be found in nupic/regions/.
-    encoders     (dict)         See createEncoder() docstring for format.
-    numCategories  (int)        Max number of categories of the input data.
-    classifierType   (str)      Specific type of classifier region, e.g. "py.SequenceClassifier";
-                                possible options can be found in nupic/regions/.
-    classifierParams   (dict)   Parameters for the model. E.g. {'maxCategoryCount': 3}                               
-                                
-  """
-  (dataSource,
-   sensorType,
-   encoders,
-   numCats,
-   classifierType, 
-   classifierParams) = args
+def validateRegions(sensor, sp, tm, classifier):
+  """ Make sure region widths fit"""
 
-  sensor = createSensorRegion(
-      network, sensorType, encoders, dataSource, numCats)
+  sensorOutputWidth = sensor.encoder.getWidth()
+  spInputWidth = sp.getSelf().inputWidth
+  spOutputWidth = sp.getSelf().columnCount
+  tmInputWidth = tm.getSelf().columnCount
+  tmOutputWidth = tmInputWidth * tm.getSelf().cellsPerColumn
 
-  sp = createSpatialPoolerRegion(network, sensor.encoder.getWidth())
+  if sensorOutputWidth != spInputWidth:
+    raise ValueError("Region widths do not fit. Sensor output width = %s. SP input width = %s"
```

Comment:
raise ValueError("Region widths do not fit. Sensor output width = {}, SP "
                      "input width = {}.".format(sensorOutputWidth, spInputWidth)